From 903e13d5357d6ecd9b0a7f093c2dfcb11ddbe3bf Mon Sep 17 00:00:00 2001
From: Jethro Beekman <jethro@jbeekman.nl>
Date: Fri, 24 May 2013 18:46:09 -0700
Subject: [PATCH 14/16] Fix lsusb -t potentially not listing all devices

There's a bug in the path-parsing code of add_usb_device(). It never checks for
a NULL terminator. This causes some devices to have a bogus
portnum/parent_portnum which in turn makes them not show in the device tree.

Here are two real-world test-cases (yes, the \0 is intentional):
Shows up: add_usb_device("1-1.5.1.4.1\0\n\241M")
Disappers: add_usb_device("1-1.5.1.4.2\0\n0c")
---
 lsusb-t.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lsusb-t.c b/lsusb-t.c
index 38a5f37..4f5b47a 100644
--- a/lsusb-t.c
+++ b/lsusb-t.c
@@ -376,7 +376,7 @@ static void add_usb_device(const char *d_name)
 		return;
 	memset(d, 0, sizeof(struct usbdevice));
 	d->busnum = i;
-	while (pn) {
+	while (*pn) {
 		p = pn + 1;
 		i = strtoul(p, &pn, 10);
 		if (p == pn)
-- 
2.4.3

